scipy/optimize/_differentialevolution.py

```diff
@@ -551,14 +586,34 @@ def __init__(self, func, bounds, args=(),
         if updating in ['immediate', 'deferred']:
             self._updating = updating
 
+        self.vectorized = vectorized
+
         # want to use parallelisation, but updating is immediate
         if workers != 1 and updating == 'immediate':
             warnings.warn("differential_evolution: the 'workers' keyword has"
                           " overridden updating='immediate' to"
                           " updating='deferred'", UserWarning)
             self._updating = 'deferred'
 
+        if vectorized and workers != 1:
+                warnings.warn("differential_evolution: the 'workers' keyword"
+                              " overrides the 'vectorized' keyword")
+                self.vectorized = vectorized = False
+
+        if vectorized and updating == 'immediate':
+            warnings.warn("differential_evolution: the 'vectorized' keyword"
+                          " has overridden updating='immediate' to updating"
+                          "='deferred'", UserWarning)
```

Comment:
Funny how it's very difficult to remember `stacklevel` when writing code, but it's easy to notice when reviewing : P

Reply:
Still needs `stacklevel`, right?